src/hotspot/share/runtime/sharedRuntime.hpp

```diff
@@ -345,6 +345,11 @@ class SharedRuntime: AllStatic {
   // compiled code.
   static methodHandle resolve_helper(bool is_virtual, bool is_optimized, TRAPS);
 
+  // MNCMNC
+  static void trigger_action_from_c1(JavaThread* current);//TRAPS);
+  static void trigger_action_from_c2(JavaThread* current);//TRAPS);
+  static void trigger_action(const char* info, TRAPS);
```

Comment:
Is it a convention to assume that the interpreter is implied if not explicitly mentioned? Is that why this function isn't called trigger_action_from_interpreter?

Reply:
Ah, I see that this function will also be called from jcmd.